devito/ir/iet/visitors.py

```diff
@@ -233,6 +135,9 @@ def _args_decl(self, args):
         """Convert an iterable of :class:`Argument` into cgen format."""
         ret = []
         for i in args:
+            if not hasattr(i, "is_ScalarArgument"):
```

Comment:
why not ``if i.is_ScalarArgument`` ?

Reply:
Because if it is not a `ScalarArgument` object, it might be an object from the `types.py` hierarchy - in which case it won't even have that flag.

Reply:
As you know, I'm not a big fan at all of this kind of ``hasattr(...)`` approach, which seems a bit hacky to me. If considered pythonic and I'm actually missing something, I'm all ears!

So I don't know what to suggest ATM, but it feels very weird that "something might not be an actual Argument``, **but**:
* other objects in other hierarchies have an analogous flag (``is_ScalarArgument``)
* below, we're using an ``ArgumentVisitor`` and this can handle stuff which are non-Arguments

what kind of objects from ``types.py`` you may encounter here?